### PR TITLE
feat:remove old zookeeper 

### PR DESCRIPTION
**Pull request by RainYuY**
## What is the purpose of the change

refer:[#13960](https://github.com/apache/dubbo/issues/13960)

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by RainYuY** [comment]
Should the first CI test fix be merged, then rerun the test case for this PR. In my own repository, it has already passed.

**Review by AlbumenJ — APPROVED** [review]

File: dubbo-remoting/dubbo-remoting-zookeeper-api/src/main/java/org/apache/dubbo/remoting/zookeeper/ZookeeperTransporter.java
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-zookeeper-api/src/main/java/org/apache/dubbo/remoting/zookeeper/ZookeeperTransporter.java]: The existence of ZookeeperTransporter is not very meaningful, so consider removing it.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-zookeeper-api/src/main/java/org/apache/dubbo/remoting/zookeeper/ZookeeperTransporter.java
Comment (Stellar1999) [reply dubbo-remoting/dubbo-remoting-zookeeper-api/src/main/java/org/apache/dubbo/remoting/zookeeper/ZookeeperTransporter.java:37]: OK, but I think I will use another PR and issue to handle this task. Because the test cases are still growing, I want to finish this PR earlier. Is that OK?

Comment (RainYuY) [reply dubbo-remoting/dubbo-remoting-zookeeper-api/src/main/java/org/apache/dubbo/remoting/zookeeper/ZookeeperTransporter.java]: Please take another look. Do you think my code idea is correct? I removed the SPI and moved the ZookeeperTransporter to the module `dubbo-remoting-zookeeper-curator5` in order to create an instance of Curator5ZookeeperTransporter. @CrazyHZM Thanks for your code review.

Comment (CrazyHZM) [reply dubbo-remoting/dubbo-remoting-zookeeper-api/src/main/java/org/apache/dubbo/remoting/zookeeper/ZookeeperTransporter.java]: @Stellar1999 ZookeeperTransporter should be renamed ZookeeperClientManager or something else. 
dubbo-remoting-zookeeper-api should also be incorporated into the curator5 module.

Comment (RainYuY) [reply dubbo-remoting/dubbo-remoting-zookeeper-api/src/main/java/org/apache/dubbo/remoting/zookeeper/ZookeeperTransporter.java]: Please take another look. Thanks for your guidance. @CrazyHZM 
If it's alright, I will fix the sample test case.

**Comment by CrazyHZM** [comment]
@Stellar1999 
We should keep the dubbo-remoting-zookeeper-curator5 module and move the dubbo-remoting-zookeeper-api module into it.

**Comment by RainYuY** [comment]
> @Stellar1999 We should keep the dubbo-remoting-zookeeper-curator5 module and move the dubbo-remoting-zookeeper-api module into it.

Please take another look. Thanks for your guidance. @CrazyHZM

**Comment by AlbumenJ** [comment]
image

**Comment by CrazyHZM** [comment]
It all looks good at the moment, but recently, CI has not been very stable. Wait until CI is fully passed before merging.
@AlbumenJ

**Comment by AlbumenJ** [comment]
Please resolve conflicts

**Pull request merged**
